tests/test_vault.py

```diff
@@ -0,0 +1,185 @@
+from ocw.lib.vault import Vault, AzureCredential, EC2Credential, GCECredential
+from datetime import datetime
+from datetime import timedelta
+import pytest
+import requests
+import webui
+import json
+import string
+import random
+import base64
+
+
+# Global test data
+namespace = 'test-qa'
+host = 'http://foo.bar'
+leases = list()
+
+
+def random_str(stringLength):
+    letters = string.ascii_letters
+    return ''.join(random.choice(letters) for i in range(stringLength))
+
+
+class MockResponse:
```

Comment:
maybe make sense to put it in separate file ? I am sure that we will use it not only here

Reply:
Yeah sure, Lets do it when we need it.